### PR TITLE
[DYN-4302 + DYN-5259] Add visual indication for deprecated packages

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

**[DYN-4302](https://jira.autodesk.com/browse/DYN-4302)**

- Disabled package install for deprecated packages from `Package Details View Extension` and added visual indications for deprecated packages such as tags and tooltips.
- Also, fixed a bug where user was still able to install packages when `Custom Package Locations` was disabled from preference settings.
- Unified UI across Package Manager window and the view extension.
- Update profile icon to a HD version

**Screenshots:**

When a package is deprecated: 

Screen Shot 2022-10-20 at 1 37 31 AM

When `Custom Package Locations` is disabled (tooltips are still visible for deprecated packages) :

Screen Shot 2022-10-20 at 1 39 15 AM

Tooltips:

Screen Shot 2022-10-20 at 1 37 41 AM
Screen Shot 2022-10-20 at 1 37 48 AM
Screen Shot 2022-10-20 at 1 37 57 AM

Profile Icon:

Screen Shot 2022-10-20 at 2 52 42 AM
Screen Shot 2022-10-20 at 2 53 59 AM

**[DYN-5259](https://jira.autodesk.com/browse/DYN-5259)**

- Fixed horizontal offset for notifications.

**Screenshot:**

Screen Shot 2022-10-20 at 2 13 35 AM
Screen Shot 2022-10-20 at 2 13 35 AM copy

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

- Disabled package install for deprecated packages from `Package Details View Extension`.
- Added visual indications for deprecated packages such as tags and tooltips.
- Fixed a bug where user was still able to install packages when `Custom Package Locations` was disabled from preference settings.
- Fixed horizontal offset for notifications.

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml [hunk 1]
```diff
@@ -559,23 +537,26 @@
...
                                                                                             </Setter.Value>
                                                                                         </Setter>
+                                                                                        <Setter Property="Opacity" Value="0.5" />
                                                                                     </DataTrigger>
-                                                                                    <DataTrigger Binding="{Binding Path=IsEnabledForInstall}" Value="False">
+                                                                                    <DataTrigger Binding="{Binding Path=CanInstall}" Value="False">
                                                                                         <Setter Property="Fill">
                                                                                             <Setter.Value>
-                                                                                                <ImageBrush ImageSource="/DynamoCoreWpf;component/UI/Images/Add - disabled.png" />
```
Comment (QilongTang) [inline src/DynamoCoreWpf/Views/PackageManager/PackageManagerSearchView.xaml:569]: Do we need to remove these related resources?

File: src/PackageDetailsViewExtension/PackageDetailItem.cs [hunk 2]
```diff
@@ -152,15 +167,18 @@ public bool CanInstall
...
+        public PackageDetailItem(string packageName, PackageVersion packageVersion, bool canInstall, bool isEnabledForInstall = true)
         {
             this.PackageName = packageName;
             this.PackageVersion = packageVersion;
             this.PackageVersionNumber = PackageVersion.version;
             this.CopyRightHolder = PackageVersion.copyright_holder;
             this.CopyRightYear = PackageVersion.copyright_year;
             this.CanInstall = canInstall;
-            
+            this.IsEnabledForInstall = isEnabledForInstall && canInstall;
```
Comment (QilongTang) [inline src/PackageDetailsViewExtension/PackageDetailItem.cs:179]: should we use this statement `isEnabledForInstall && canInstall` in the getter instead so the value is always up to date?

**Comment by QilongTang** [comment]
Regression: 
[DynamoCoreWpfTests.PackageDetailsViewExtensionTests.TestLinksDisplayedInView](https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/7335/testReport/junit/DynamoCoreWpfTests/PackageDetailsViewExtensionTests/TestLinksDisplayedInView/)

File: src/PackageDetailsViewExtension/PackageDetailItem.cs (hunk 2 above)
Comment (zeusongit) [reply src/PackageDetailsViewExtension/PackageDetailItem.cs:179]: Do you mean like this:
```
public bool IsEnabledForInstall
{
    get => isEnabledForInstall && canInstall;
    set
    {
        isEnabledForInstall = value;
        RaisePropertyChanged(nameof(IsEnabledForInstall));
    }
}
```
and set like:
```
this.IsEnabledForInstall = isEnabledForInstall;
```

Comment (QilongTang) [reply src/PackageDetailsViewExtension/PackageDetailItem.cs:179]: On a second thought, if the setter and getter are pointing to different values, it is also confusing. I guess I am just confused that we have two booleans which are very similar, maybe name them slightly more obvious to each case?

Comment (zeusongit) [reply src/PackageDetailsViewExtension/PackageDetailItem.cs:179]: I actually used the same variable names to be consistent with the PackageManagerSearch window logic,
`canInstall` handles if the package is already loaded and `IsEnabledForInstall ` handles preference settings toggle value.

File: src/PackageDetailsViewExtension/PackageDetailItem.cs [hunk 3]
```diff
@@ -134,6 +135,10 @@ public List<string> Packages
             }
         }
 
+        /// <summary>
+        /// Returs true if package version is not already installed,
+        /// true if not installed
```
Comment (QilongTang) [inline src/PackageDetailsViewExtension/PackageDetailItem.cs:140]: Is the second line duplicate here?

**Review by QilongTang — APPROVED** [review]
LGTM, once conflicts addressed and PR checks pass, good to merge

**Comment by mjkkirschner** [comment]
Sorry to comment at the end of this PR and I assume it's already been discussed but why block downloads for deprecated packages? Isn't this annoying if a user needs that package?
For example - nuget does not block downloads for deprecated packages - it does hide them though.

**Comment by zeusongit** [comment]
Downloads for deprecated packages was already blocked, what we did in this PR is just to block it in the package details view extension as well.

**Pull request merged**